server/src/main/java/org/candlepin/resource/HypervisorResource.java

```diff
@@ -152,11 +154,16 @@ public HypervisorUpdateResultDTO hypervisorUpdate(
         @QueryParam("create_missing") @DefaultValue("true") boolean createMissing) {
         log.debug("Hypervisor check-in by principal: {}", principal);
 
-        if (hostGuestDTOMap == null) {
+        Map<String, List<GuestIdDTO>> convertedHostGuestIdDTOMap;
+
+        if (hostGuestMap == null) {
             log.debug("Host/Guest mapping provided during hypervisor checkin was null.");
             throw new BadRequestException(
                 i18n.tr("Host to guest mapping was not provided for hypervisor check-in."));
         }
+        else {
+            convertedHostGuestIdDTOMap = convertMapToGuestIdDTO(hostGuestMap);
```

Comment:
In case we do want to change the API to have a list of Strings instead of GuestIdDTOs, then there is no need to do any conversion here. Instead we could update the code to process List of Strings, and change `GuestIdResource.populateEntities` to the conversion to a GuestId object.

Reply:
Updated as per suggestion.